certbot/tests/util.py

```diff
@@ -379,3 +380,26 @@ def hold_lock(cv, lock_path):  # pragma: no cover
     cv.notify()
     cv.wait()
     my_lock.release()
+
+
+class MockInstallerAutoHSTS(enhancements.AutoHSTSEnhancement):
```

Comment:
I'm not sure if we need this class. I think we can get by with setting the spec attribute of the [Mock class](https://docs.python.org/3/library/unittest.mock.html#the-mock-class) to `enhancements.AutoHSTSEnhancement` so it'll pass `isinstance` checks. Is there something else we get here?

Reply:
But what's wrong with `installer.update_autohsts.call_count`, `installer.update_autohsts.called`, `installer.update_autohsts.call_args`, etc?

This seems more conventional and takes less code. The only change with the current approach I see is you access these values on `installer.update_counter` rather than `installer.update_autohsts`.

Reply:
You have a point, and the tests are now refactored. The tests for `interfaces.GenericUpdater` and `interfaces.RenewDeployer` were similar and handled in the same files, so I went ahead and refactored those too for consistency.